external/mmsegmentation/segmentation_tasks/apis/segmentation/nncf_task.py

```diff
@@ -186,7 +186,9 @@ def optimize(
         else:
             update_progress_callback = default_progress_callback
 
-        time_monitor = TrainingProgressCallback(update_progress_callback)
+        time_monitor = OptimizationProgressCallback(update_progress_callback,
```

Comment:
`optimization_parameters.update_progress` is not of `UpdateProgressCallback` type.

Reply:
Technically yes, but everywhere else it used as if it is, e.g.:
https://github.com/openvinotoolkit/training_extensions/blob/2bbc890c4d9c70ceb02108e84750a12ae207a081/ote_sdk/ote_sdk/entities/train_parameters.py#L59
or
https://github.com/openvinotoolkit/training_extensions/blob/2bbc890c4d9c70ceb02108e84750a12ae207a081/external/mmsegmentation/segmentation_tasks/apis/segmentation/train_task.py#L76

Probably because one can implicitly be cast to another via `typing.cast()`, e.g. `cast(UpdateProgressCallback, default_progress_callback)` will execute fine.

Reply:
>  everywhere else it is used as if it is

And this is the problem. This code is simply incorrect. Please fix.

Reply:
@druzhkov-paul Let's fix it in another PR because the fix affects the training task as well.

Reply:
@alexsu52 It's not just a cosmetic imperfection here. Callback of a wrong type is passed into here and then called with incorrect number/type of arguments. Tweaking tests to keep silence is not a proper way to solve this. So, please address this issues and run SC E2E and BMM tests.